src/imitation/model_env.py

```diff
@@ -9,40 +9,56 @@
 
 
 class ModelBasedEnv(gym.Env, abc.ABC):
```

Comment:
Nitpick: none of the abstract methods in this class actually require known dynamics. All you need is a re-settable simulator, which is a much weaker condition that all current Gym environments could satisfy anyway. We could just as easily use this class to wrap Gym, Atari, etc. environments for non-model-based algorithms like vine-mode TRPO, which needs the ability to "branch" from states. Maybe a name like `ResettableEnv` or `SimulatorEnv` better reflects the generality of this interface?